fe1-web/model/objects/WalletCryptographyHandler.ts

```diff
@@ -15,62 +30,134 @@ export class WalletCryptographyHandler {
+  public async encrypt(plaintext: Uint8Array): Promise<ArrayBuffer> {
+    const key: CryptoKey = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
+    if (key === undefined) {
+      throw Error('Error while retrieving encryption key from database: undefined');
+    }
+    // @ts-ignore
+    const ciphertext = await this.cryptography.subtle.encrypt(this.algorithm, key, plaintext);
+    return ciphertext;
   }
 
   /**
-   * decrypts the encrypted ed25519 token with the RSA key stored in the indexedDB database
-   * @param encryptedToken ed25519 encrypted token (ArrayBuffer)
+   * decrypts the encrypted ed25519 seed with the RSA key stored in the indexedDB database
+   * @param ciphertext ed25519 encrypted seed (ArrayBuffer)
    */
-  public async decrypt(encryptedToken: ArrayBuffer): Promise<ArrayBuffer> {
+  public async decrypt(ciphertext: ArrayBuffer): Promise<ArrayBuffer> {
     const key = await this.getKeyFromDatabase(STRINGS.walletPrivateKey);
-    const plaintextToken = await window.crypto.subtle.decrypt(this.algorithm, key, encryptedToken);
-    return plaintextToken;
+    if (key === undefined) {
+      throw Error('Error while retrieving decryption key from database: undefined');
+    }
+    // @ts-ignore
+    const plaintext = await this.cryptography.subtle
+      .decrypt(this.algorithm, key, ciphertext);
+    return plaintext;
+  }
+
+  /**
+   * adds an encryption/decryption key to the storage
+   * @private
+   * @param key the key that will be used to encrypt/decrypt all the
+   * seed generating all tokens in the wallet
+   */
+  private async putKeyInDatabase(key: WalletCryptoKey): Promise<void> {
+    await set(this.publicKeyId, key.publicKey);
+    await set(this.privateKeyId, key.privateKey);
   }
 
   /**
-   * This function retrieves the RSA key from database
-   * @param type 'public' if the desired key is the public key,
-   * 'private' if the desired key is the private key
+   * returns the requested encryption key from the storage (only one crypto-key allowed in storage)
+   * @private
+   * @param type 'public' if the desired key is the public (encryption) key,
+   * 'private' if the desired key is the private (decryption) key
    */
-  private getKeyFromDatabase: (type: string) => Promise<CryptoKey> =
-  async (type: string) => {
-    const key: CryptoKey = await this.keyDatabase.getKey(type);
+  private async getKeyFromDatabase(type: string): Promise<CryptoKey> {
+    if (type === STRINGS.walletPrivateKey) {
+      const key = await get(this.privateKeyId);
+      return key;
+    }
+    const key = await get(this.publicKeyId);
     return key;
-  };
+  }
+
+  /**
+   * this is used to update the encryption key in case it will ever be
+   * necessary to change it, it updates the key since only one crypto-key
+   * is allowed in the wallet storage.
+   * @private
+   * @param key the new key for encryption/decryption
+   */
+  private async updateKeyInDatabase(key: WalletCryptoKey): Promise<void> {
```

Comment:
typically this would be useful if we need to restore the key from a backup, excellent! 👍